test/over_react/component_declaration/builder_integration_tests/new_boilerplate/null_safety_validate_required_props_test.dart

```diff
@@ -0,0 +1,233 @@
+// Copyright 2023 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react/over_react.dart';
+import 'package:react_testing_library/react_testing_library.dart' as rtl;
+import 'package:test/test.dart';
+
+import '../../../../test_util/test_util.dart';
+
+part 'null_safety_validate_required_props_test.over_react.g.dart';
+
+void main() {
+  group('(New boilerplate) validates required props:', () {
+    group('non-nullable required prop', () {
+      group('throws when a prop is required and not set', () {
+        test('on mount', () {
```

Comment:
I think we want to verify more specifically that these throw when the props are invoked, before they're even attempted to be rendered.

So the setup should be more like:
```dart
          expect(() {
            (ComponentTest()
              ..requiredNullable = true
            )();
          },
              throwsA(isA<MissingRequiredPropsError>()...
```

Though I do think that also having test coverage that they render properly, like you have now, is probably a good idea.